hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/ws/WsCommands.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.cmdgw.command.ws;
+
+import javax.ejb.Singleton;
+
+import org.hawkular.bus.common.BasicMessage;
+import org.hawkular.cmdgw.annotations.Ws;
+import org.hawkular.cmdgw.api.EchoRequest;
+import org.hawkular.cmdgw.api.ResourcePathDestination;
+import org.hawkular.cmdgw.api.UiSessionDestination;
+import org.hawkular.cmdgw.command.Command;
+import org.hawkular.cmdgw.command.Commands;
+import org.hawkular.cmdgw.command.NoCommandForMessageException;
+
+/**
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+@Singleton @Ws
+public class WsCommands implements Commands {
+
+    private final ResourcePathDestinationWsCommand resourcePathDestinationWsCommand = //
+    new ResourcePathDestinationWsCommand();
+
+    private final EchoCommand echoCommand = new EchoCommand();
+
+    private final UiSessionDestinationWsCommand uiSessionDestinationWsCommand = new UiSessionDestinationWsCommand();
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <REQ extends BasicMessage, RESP extends BasicMessage> Command<REQ> getCommand(Class<REQ> requestClass)
```

Comment:
They shouldn't store state, but that's not the only way concurrency issues could crop up. I realize that current implementations are thread safe, but are we sure in the future that is always going to be the case? For now, we can leave it, but put a REALLY LOUD Javadoc warning stating command impls need to be thread safe. In the future if we find they are not thread safe, we can always change it to be instantiated. But let it be known - I said it here first - concurrency risks abound here. (that's so I can say "I told you so" later :-)